src/uu/tee/src/tee.rs

```diff
@@ -367,3 +385,44 @@ impl Read for NamedReader {
         }
     }
 }
+
+/// Check that if stdout is a pipe, it is not broken.
+#[cfg(target_os = "linux")]
+pub fn ensure_stdout_not_broken() -> Result<bool> {
+    use nix::{
+        poll::{PollFd, PollFlags},
+        sys::stat::{fstat, SFlag},
+    };
+    use std::os::fd::{AsFd, AsRawFd};
+
+    let out = stdout();
+
+    // First, check that stdout is a fifo and return true if it's not the case
+    let stat = fstat(out.as_raw_fd())?;
+    if !SFlag::from_bits_truncate(stat.st_mode).contains(SFlag::S_IFIFO) {
+        return Ok(true);
+    }
+
+    // POLLRDBAND is the flag used by GNU tee.
+    let mut pfds = [PollFd::new(out.as_fd(), PollFlags::POLLRDBAND)];
+
+    // Then, ensure that the pipe is not broken
+    let res = nix::poll::poll(&mut pfds, Option::<u8>::None)?;
```

Comment:
1. Maybe `PollTimeout::None` might be clearer?
2. Can you say more about what situation would cause this `poll` to wait forever, if it is possible?

Reply:
1. Will do, Thank for noticing this
2. There's no reason that poll would hang, but still we need to provide it some timeout argument, either it is None or Some. It's just the way the return value of poll should be interpreted